key-manager/grant-permissions.ts

```diff
@@ -0,0 +1,86 @@
+import { ethers } from 'ethers';
+import { ERC725 } from '@erc725/erc725.js';
+import LSP6Schema from '@erc725/erc725.js/schemas/LSP6KeyManager.json';
+import UniversalProfileArtefact from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
+
+const myUniversalProfileAddress = process.env.UP_ADDR || '';
+
+const RPC_ENDPOINT = 'https://4201.rpc.thirdweb.com';
+const provider = new ethers.JsonRpcProvider(RPC_ENDPOINT);
+
+// Initialize erc725.js with the permission data keys from LSP6 Key Manager
+const erc725 = new ERC725(
+  LSP6Schema,
+  myUniversalProfileAddress,
+  RPC_ENDPOINT,
+  {},
+);
+
+// Create the permissions for the beneficiary address
+const beneficiaryPermissions = erc725.encodePermissions({
+  SETDATA: true,
+});
+
+// EOA address of the new controller
+const myBeneficiaryAddress = '0xcafecafecafecafecafecafecafecafecafecafe';
+
+// Retrieve the current controllers of the Universal Profile
+const addressPermissionsArray = await erc725.getData('AddressPermissions[]');
+let currentControllerAddresses = null;
+let currentControllerLength: number = 0;
```

Comment:
Type will be inferred automatically with the initial value

```suggestion
let currentControllerLength = 0;
```